cost/aws/old_snapshots/aws_delete_old_snapshots.pt

```diff
@@ -229,15 +299,43 @@ escalation "esc_delete_snapshot" do
       description "Expain why are you approving the action"
     end
   end
-  run "delete_snapshot",data
+  run "take_action",data
 end
 
 ###############################################################################
 # Cloud Workflow
 ###############################################################################
 
- define delete_snapshot($data) do
- foreach $item in $data do
+define take_action($data) return $all_responses do
+  $all_responses = []
+  foreach $item in $data do
+    call deregisterImageFromSnapshot($item)
```

Comment:
I'm concerned about deregistering all the images from snapshot.  What about adding a new parameter to ask to Deregister the image.

Reply:
Yes, @cdwilhelm we can take the input from the end-user whether to deregistering the image or not. If Yes we can deregister the image and then delete the snapshot. In the case of No, we need to skip the deletion of snapshot because without deregister we can't delete the snapshot directly.

We can mention in the input parameter description that 
    - 1. If Yes, deregister the image and delete the snapshot.
    - 2. If No, We should skip all the snapshots linked with the image.

Reply:
@cdwilhelm what do u think on above-made comments?

Reply:
I'm thinking a little different approach that will work with the new selectable resources feature.  The new approach would be to have two escalations/actions.  1) deregister image, 2) delete snapshot.   No extra parameter is needed.

We can test now with the API or wait until the UI is ready sometime in Jun.

Reply:
@SheikSena   I was looking at this to determine it's state.  It looks like it's not ready to use selectable actions or take a parameter to deregister the ami's.  There is already a ticket to do selectable actions, but it doesn't include the degister the ami action.  
Can we add the param to degister the ami  and update the CWF so we can release.  Then we can merge with the selectable action PR.  what do you think?